src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -414,6 +419,70 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       particleGun->GeneratePrimaryVertex(anEvent);
     }//old muline format
   }//useMuLineEvt
+  
+  else if(useAmBeEvt){ // Diego Costas (diego.costas.rodriguez@usc.es) 2023
+    // Create the energy distribution object
+    G4SPSEneDistribution* nEnergyDist = new G4SPSEneDistribution();
+    G4SPSRandomGenerator* rGen = new G4SPSRandomGenerator();
+
+    // Set the neutron energy distribution from a file
+    nEnergyDist->SetEnergyDisType("Arb");
+    nEnergyDist->ArbEnergyHistoFile("../data/resampled_nSpectrum.txt");
+    nEnergyDist->SetBiasRndm(rGen);
+    nEnergyDist->ArbInterpolate("Lin");
+
+    // Obtain the definition of the neutron and sample an energy from the distribution
+    G4ParticleTable* particleTable = G4ParticleTable::GetParticleTable();
+    G4ParticleDefinition* neutronDefinition = particleTable->FindParticle("neutron");
+    //G4ParticleDefinition* neutronDefinition = G4Neutron::Definition();
+    G4double nEnergy = nEnergyDist->GenerateOne(neutronDefinition);
+
+    // Define PDG IDs, energies and desired particle names
+    G4int pdgids[] = {2112, 22};
+    G4double energys[] = {nEnergy, 0.0};
+    G4String particleNames[] = {"neutron", "gamma"};
+
+    // Definir las probabilidades correspondientes a cada energía de partícula gamma
+    std::vector<double> probabilities = {0.26, 0.65, 0.08};
+    std::vector<double> energies = {0.0, 4.4, 7.7};
+
+    // Generar un número aleatorio ponderado para seleccionar la energía de la partícula gamma
+    std::random_device rd;
+    std::mt19937 gen(rd());
```

Comment:
From what I can tell, this creates a random number `rd` (uniformly-distributed integer random number that produces non-deterministic random numbers.), which is then used for the seed for the `mersenne twister engine` gen

Using the `mt19937` is fine. However it seems to be non-repeatable. Can you, instead of using `random_device`, create a new .mac file option in `WCSimPrimaryGeneratorMessenger` that is the AmBe source seed? You'll again need to modify the code to only create `gen` once, on the first event